src/lib/device_tree/src/lib.rs

```diff
@@ -185,11 +185,19 @@ pub struct FdtIterator<'a, D: Driver> {
 impl<'a, D: Driver> FdtIterator<'a, D> {
     #[allow(clippy::should_implement_trait)]
     pub fn next<'b>(&'b mut self) -> Option<Result<Entry<'b, SectionReader<'a, SectionReader<'a, D>>>>> {
```

Comment:
Would it make the error handling simpler if you swap the return to a `Result<Option<...`? Might not be possible if it has to match a certain signature.

Reply:
We do not have to match any signature (we stopped using standard iterator trait here).

I wasn't sure if it will be ergonomic for the users of the iterator if we swap the return type, but it seems it is actually even better than before. Added a new commit to swap the return type.